hypothesis-python/src/hypothesis/errors.py

```diff
@@ -117,6 +117,13 @@ class HypothesisDeprecationWarning(HypothesisWarning, FutureWarning):
     """
 
 
+class HypothesisSideeffectWarning(HypothesisDeprecationWarning):
```

Comment:
```suggestion
class HypothesisSideeffectWarning(HypothesisWarning):
```
The warning should _not_ be a deprecation warning (per my branch 😅); we'll want to leave this as a warning ~forever so that users can still use side-effect-ful plugins and -Wignore or -Werror as they prefer